### PR TITLE
[STC-2] Users can reset their password

### PR DESCRIPTION
**Pull request by jormsby2**
Added reset request view and controller
Added reset request success view and controller
Added password change view and controller
Added link to reset request view on layouts>main.html

### CODE REVIEW

File: scriptoocloud/pom.xml [hunk 1]
```diff
@@ -14,11 +14,11 @@
...
-        
-        <db.driver>com.mysql.jdbc.Driver</db.driver>
-        <db.database.url>jdbc-url-to-db</db.database.url>
-        <db.user>db-user</db.user>
-        <db.password>db-password</db.password>
+         
+        <db.driver>com.mysql.cj.jdbc.Driver</db.driver>
+        <db.database.url>jdbc:mysql://localhost:3306/test</db.database.url>
+        <db.user>root</db.user>
+        <db.password>root</db.password>
```
Comment (jdamerow) [inline scriptoocloud/pom.xml:21]: try avoiding pushing your local configurations to GitHub. You can use Maven's settings.xml file instead.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/config/SimpleUsersConfig.java [hunk 2]
```diff
@@ -9,6 +9,19 @@
...
-        
+    /*
+         simpleUsers.emailUsername("");
+         simpleUsers.emailPassword("");
+         simpleUsers.emailServerHost("");
+         simpleUsers.emailServerPort("");
+  		 simpleUsers.emailFrom("");
+  		 simpleUsers.instanceUrl(""); 
+         simpleUsers.appName("");
+     */
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/config/SimpleUsersConfig.java:20]: remove if not needed; also, you probably want to put email configurations into a config file and use the configurations here (`@Value`, `@PropertySource`)

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/manager/AdminController.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package edu.asu.diging.scriptoocloud.web.manager;
+
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class AdminController{
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/manager/AdminController.java:6]: what is this controller for?

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordChangeController.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import edu.asu.diging.simpleusers.web.forms.UserForm;
+ 
+@Controller
+public class PasswordChangeController{
+
+ 	@RequestMapping(value = "/password/change", method=RequestMethod.GET)
+    public String get(Model model) {
+		model.addAttribute("user", new UserForm());
+		return "/password/change";	
+    }
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordChangeController.java:17]: formatting

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordResetRequestController.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import edu.asu.diging.simpleusers.web.forms.UserForm;
+
+@Controller
+public class PasswordResetRequestController {
+
+    @RequestMapping(value = "/reset", method=RequestMethod.GET)
+    public String get(Model model) {
+		model.addAttribute("user", new UserForm());
+		return "/reset/request";	
+    }
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordResetRequestController.java:17]: indentation

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordResetSentController.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+ 
+@Controller
+public class PasswordResetSentController{
+	
+   @RequestMapping(value = "/reset/sent", method=RequestMethod.GET)
+    public String reset(Model model) {
+    	return "/reset/sent";
+    }
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordResetSentController.java:14]: formatting

File: scriptoocloud/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 7]
```diff
@@ -63,6 +63,9 @@
 	          	<li sec:authorize="isAnonymous()" role="presentation">
 	         	 	<a title="Sign Up" th:href="@{/register}"><i class="fas fa-user-plus"></i></a>
 	          	</li>
+	          	<li sec:authorize="isAnonymous()" role="presentation">
+	         	 	<a title="Reset Password" th:href="@{/reset}"><i class="fas fa-user-plus"></i></a>
```
Comment (jdamerow) [inline scriptoocloud/src/main/webapp/WEB-INF/views/layouts/main.html:67]: is that the same icon as for registering? Maybe use a different one ;)

**Review by jdamerow — CHANGES_REQUESTED** [review]
Just in case you need a simple email set up for testing this, take a look at "fakeSMTP".

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordChangeController.java (hunk 4 above)
Comment (jcoro) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/password/PasswordChangeController.java:17]: Hey Jason, I figured out the implementation.  This Controller is not needed in the application (the Controller from simple-users takes care of Post and Get). See my comment on ../password/change.html as well.

File: scriptoocloud/src/main/webapp/WEB-INF/views/password/change.html [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    <title>Authority Matcher</title>
+  </head>
+  <body>
+  <div layout:fragment="content">
+  <h2>Change password</h2>
+  <form action="#" th:action="@{/password/change}" method="POST" th:object="${user}">
+		<input type="hidden" name="${_csrf.parameterName}" value="${_csrf.token}" />
+		
+		<label>New Password:</label>
+		<input type="password" th:field="*{password}" class="form-control input-sm"></password>
```
Comment (jcoro) [inline scriptoocloud/src/main/webapp/WEB-INF/views/password/change.html:12]: The key for this file is that it doesn't need th:object =${user} as the Professor said.  The real key is though, to change the th:field(s) to th:value instead.  If it sees th:field it expects a model object, I believe.

**Pull request closed without merging**